lib/bolt/node/winrm.rb

```diff
@@ -337,7 +518,23 @@ def _run_task(task, input_method, arguments)
           # NOTE: cannot redirect STDIN to a .ps1 script inside of PowerShell
           # must create new powershell.exe process like other interpreters
           # fortunately, using PS with stdin input_method should never happen
-          output = execute("try { &""#{remote_path}"" } catch { exit 1 }")
+          output = execute(<<-PS)
+& { # create a new scope
+  $_task_path = [Text.Encoding]::UTF8.GetString(
+    [Convert]::FromBase64String('#{Base64.encode64(remote_path)}')
+  )
+  $_task_args = Get-ContentAsJson (
```

Comment:
Nit - `$_task_args` and `$_task_path` are not idiomatic PowerShell variable names, which tends to prefer camelCase rather than snake_case.

Reply:
How about the leading underscore? (I used it deliberately so as to make the variables unlikely to conflict with any variables in the task.)

Reply:
Specifically: Would `$_taskPath` & `$_taskArg` be ok?

Reply:
Ah, I understand why you did this now.  Per my other comment at https://github.com/puppetlabs/bolt/pull/179#discussion_r155053986, I don't think you need `$_task_path` assigned to a variable (but double check my comment there).

If you want to prevent the child scope inheriting from the parent you can use `$private:foo = 'bar'` (the long form is `New-Variable -Name foo -Value 'bar' -Option private`)

```
PS /Users/Iristyle/source/bolt> echo $foo
PS /Users/Iristyle/source/bolt> cat ./foo.ps1
Write-Host $foo
Write-Host $script:foo
Write-Host $global:foo
PS /Users/Iristyle/source/bolt> echo $foo
PS /Users/Iristyle/source/bolt> $private:foo = 'bar'
PS /Users/Iristyle/source/bolt> echo $foo
bar
PS /Users/Iristyle/source/bolt> ./foo.ps1



PS /Users/Iristyle/source/bolt> & ./foo.ps1



```

Reply:
$private:taskArgs is the winner ...